staging/src/k8s.io/apiserver/pkg/storage/cacher/cache_watcher.go

```diff
@@ -454,6 +454,10 @@ func (c *cacheWatcher) processInterval(ctx context.Context, cacheInterval *watch
 	const initProcessThreshold = 500 * time.Millisecond
 	startTime := time.Now()
 
+	if cacheInterval.resourceVersion > resourceVersion {
```

Comment:
please make sure it is valid for all cases.

Reply:
Yes, this is fine.
But please add a comment like that:
```
// cacheInterval may be created from version being more fresh than requested
// (e.g. for NotOlderThan semantic). In such case, we need to prevent watch event
// with lower resourceVersion from being delivered to ensure watch contract.
```